src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditManager.cs

```diff
@@ -121,9 +121,7 @@ public void Dispose(Exception abortException)
                 {
                     while (_waiters.TryDequeue(out Waiter waiter))
                     {
-                        waiter.TrySetException(abortException != null ? (Exception)
-                            new IOException(SR.net_http_request_aborted, abortException) :
-                            CreateObjectDisposedException(forActiveWaiter: true));
+                        waiter.TrySetException(CreateObjectDisposedException(forActiveWaiter: true));
```

Comment:
I've not yet read most of the PR yet, but... is this revert a mistake, or is it no longer needed because we now expect there never to be a waiter present when a CreditManager is disposed?  If the latter, seems there should be a Debug.Fail inside the waiters != null block.

Reply:
We now expect there never to be a waiter present. 

Re Debug.Fail, where do you want this?

Reply:
> Re Debug.Fail, where do you want this?

As the first line inside `if (_waiters != null)` block.

Reply:
Ah, yeah. Sure, I can make that change.